File: springboot-mall/src/main/java/com/tonyydl/springbootmall/model/Product.java
package com.tonyydl.springbootmall.model;

import com.tonyydl.springbootmall.constant.ProductCategory;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(toBuilder = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    private Integer productId;
    private String productName;
    private ProductCategory category;
    private String imageUrl;
    private Integer price;
    private Integer stock;
    private String description;
    private Date createdDate;
    private Date lastModifiedDate;
}

File: android-app/app/src/main/java/com/tonyydl/shoppingmallapp/data/vo/Product.kt
package com.tonyydl.shoppingmallapp.data.vo

import com.tonyydl.shoppingmallapp.data.ProductCategory

data class Product(
    val productId: Int,
    val productName: String,
    val category: ProductCategory,
    val imageUrl: String,
    val price: Int,
    val stock: Int,
    val description: String,
    val createdDate: Long,
    val lastModifiedDate: Long
)